diff -p -up xf86-input-hidtouch-9.04.04/src/hidtouch__body.h.xiABI3 xf86-input-hidtouch-9.04.04/src/hidtouch__body.h
--- xf86-input-hidtouch-9.04.04/src/hidtouch__body.h.xiABI3	2009-04-04 09:07:54.000000000 +0200
+++ xf86-input-hidtouch-9.04.04/src/hidtouch__body.h	2010-03-23 11:09:44.000000000 +0100
@@ -171,7 +171,11 @@ hdtOnDeviceInit__initAxes(DeviceIntPtr d
 
 
     if (!InitValuatorClassDeviceStruct(device,
-            num_axes, GetMotionHistory, GetMotionHistorySize(),
+            num_axes,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+            GetMotionHistory,
+#endif
+            GetMotionHistorySize(),
             Absolute))
     {
         return BadAlloc;
